### PR TITLE
feat: add `ibc-crosschain` module

### PR DESCRIPTION
**Pull request by lumtis**
# Description

- Add `ibc-crosschain` module. The module is currently empty but take ibcTransfer and crosschain modules as a dependencies for future work

- Add placeholders in `ibc-crosschain` to implement the IBCModule interface

- Add `ibc-crosschain` module as a dep to `crosschain`. Since there is a circular dep, crosschain must be first initialized, passed to ibc-crosschain when initialized, the call the `SetIBCTransferKeeper` in crosschain to set the ibc-crosschain keeper

- Update testkeeper package to add capability, ibc and ibc-transfer modules to SDK modules, add ibc-crosschain module and update crosschain dep

Closes: https://github.com/zeta-chain/node/issues/2128

### CODE REVIEW

File: testutil/keeper/mocks/crosschain/authority.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v2.38.0. DO NOT EDIT.
+// Code generated by mockery v2.39.1. DO NOT EDIT.
```
[the remark this answers is not in the document]
Comment (lumtis) [reply testutil/keeper/mocks/crosschain/authority.go:1]: Yes, I added an issue for this in the past: https://github.com/zeta-chain/node/issues/1882
I will put it for the milestone, might be trivial

File: app/app.go [hunk 2]
```diff
@@ -207,6 +211,7 @@ var (
 		authoritymodule.AppModuleBasic{},
 		lightclientmodule.AppModuleBasic{},
 		crosschainmodule.AppModuleBasic{},
+		ibccrosschain.AppModule{},
```
[the remark this answers is not in the document]
Comment (lumtis) [reply app/app.go]: Mistake 6ddf23b9

File: docs/cli/zetacored/zetacored_query.md [hunk 3]
```diff
@@ -43,6 +43,7 @@ zetacored query [flags]
 * [zetacored query group](zetacored_query_group.md)	 - Querying commands for the group module
 * [zetacored query ibc](zetacored_query_ibc.md)	 - Querying commands for the IBC module
 * [zetacored query ibc-transfer](zetacored_query_ibc-transfer.md)	 - IBC fungible token transfer query subcommands
+* [zetacored query icrosschain](zetacored_query_icrosschain.md)	 - Querying commands for the icrosschain module
```
[the remark this answers is not in the document]
Comment (lumtis) [reply docs/cli/zetacored/zetacored_query.md:46]: Ok, I just realize the module name is used in the doc so we should use something clearer. `ibc-` can't be used neither the module name of ibc transfer is `transfer`

We can also refactor the doc generation logic but might be more complex.

Maybe I can use `zetaibccrosschain` for now

[the remark this answers is not in the document]
Comment (lumtis) [reply docs/cli/zetacored/zetacored_query.md]: 96837523

File: testutil/keeper/crosschain.go [hunk 4]
```diff
@@ -23,28 +24,61 @@ import (
...
-	UseAuthorityMock   bool
-	UseLightclientMock bool
+	UseBankMock          bool
+	UseAccountMock       bool
+	UseStakingMock       bool
+	UseObserverMock      bool
+	UseFungibleMock      bool
+	UseAuthorityMock     bool
+	UseLightclientMock   bool
+	useIBCCrosschainMock bool
```
[the remark this answers is not in the document]
Comment (lumtis) [reply testutil/keeper/crosschain.go]: c7a07d7e

File: testutil/keeper/crosschain.go [hunk 5]
```diff
@@ -148,6 +173,36 @@ func CrosschainKeeperWithMocks(
 		lightclientKeeper,
 	)
 
+	// initialize ibccrosschain keeper and set it to the crosschain keeper
+	// there is a circular dependency between the two keepers, crosschain keeper must be initialized first
```
[the remark this answers is not in the document]
Comment (lumtis) [reply testutil/keeper/crosschain.go:177]: 314127c6

File: x/ibccrosschain/module.go [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+func (AppModuleBasic) RegisterRESTRoutes(_ client.Context, _ *mux.Router) {
+}
+
+// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the module.
+func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, _ *runtime.ServeMux) {
+	err := types.RegisterClientCtx(clientCtx)
+	if err != nil {
+		fmt.Println("RegisterQueryHandlerClient err: %w", err)
+	}
+	// TODO: Uncomment once query created
```
[the remark this answers is not in the document]
Comment (lumtis) [reply x/ibccrosschain/module.go:81]: I'm not sure we're actually ever going to implement a query for this module so not sure what to put in the issue

File: x/ibccrosschain/client/cli/query.go [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/spf13/cobra"
+	"github.com/zeta-chain/zetacore/x/ibccrosschain/types"
+)
+
+// GetQueryCmd returns the cli query commands for this module
```
Comment (kingpinXD) [inline x/ibccrosschain/client/cli/query.go:11]:
```suggestion
// GetQueryCmd returns the cli query commands for ibccrosschain module
```

File: x/ibccrosschain/client/cli/tx.go [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/spf13/cobra"
+	"github.com/zeta-chain/zetacore/x/ibccrosschain/types"
+)
+
+// GetTxCmd returns the transaction commands for this module
```
Comment (kingpinXD) [inline x/ibccrosschain/client/cli/tx.go:11]:
```suggestion
// GetTxCmd returns the transaction commands for ibccrosschain module
```

**Review by kingpinXD — APPROVED** [review]
lgtm

**Pull request merged**